nixos/services/mail/default.nix

```diff
@@ -218,7 +233,27 @@ in {
               inherit (role) mailHost webmailHost;
             });
           } config.fclib.network.fe.dualstack.addressesQuoted);
-        in listToAttrs (map cfgForDomain (attrNames (filterAttrs (domain: config: config.enable && config.autoconfig) role.domains)));
+        in listToAttrs ((map cfgForDomain (attrNames (filterAttrs (domain: config: config.enable && config.autoconfig) role.domains))) ++
+          (optional (role.imprintUrl != null || role.imprintText != null ) (lib.nameValuePair role.mailHost
+            (lib.mkMerge [
+              {
```

Comment:
listen addresses must be set the the frontend IPs here or Nginx will prefer the autoconfig vhosts (they are already listening on fe only)